### PR TITLE
Add fail2ban configuration (closes #39)

### LINKED ISSUE #39: Set fail2ban rules
In order to protect against brute-force attacks, fail2ban rules should ideally be set. There's an example [here](https://www.c-rieger.de/nextcloud-installation-guide/#c103) that can serve as guideline.

### PR DESCRIPTION
**Pull request by JimboJoe**
## Problem
- *This application could benefit from fail2ban*

## Solution
- *Add fail2ban configuration (following work on #40) *

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [x] **Code review** : Maniack C
- [x] **Approval (LGTM)** : Maniack C
- [x] **Approval (LGTM)** : Kay0u
- **CI succeeded** : 
[Build Status](https://ci-apps-dev.yunohost.org/jenkins/job/nextcloud_ynh%20PR182/)
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by maniackcrudelis** [comment]
I have to ask... Did you test it ?

**Comment by maniackcrudelis** [comment]
An interesting argument with --failregex to test the robustness of getopts by the way.
I confirm that it works perfectly, the string is rendered as ask into the config file.

**Comment by JimboJoe** [comment]
Yep, was tested OK (as in the checkbox :stuck_out_tongue: ) :+1:

File: scripts/restore [hunk 1]
```diff
@@ -141,6 +141,23 @@ ynh_multimedia_build_main_dir
...
+ynh_restore_file "/etc/fail2ban/filter.d/$app.conf"
+
+# Make sure a log file exists (mostly for CI tests)
+logfile="/home/yunohost.app/$app/data/nextcloud.log"
+if [ ! -f "$logfile" ]; then
+	touch "$logfile"
+	chown $app: "$logfile"
+fi
+
+ynh_systemd_action --action=restart --service_name=fail2ban
```
Comment (kay0u) [inline scripts/restore]: CI failed `line 159:  ynh_systemd_action: command not found`

**Comment by JimboJoe** [comment]
Will be merged in 3 days.

**Pull request merged**